cg/test/subjects/data-shares.api.spec.js

```diff
@@ -0,0 +1,140 @@
+jest.mock('../../src/utils/blockchain/web3-provider-factory');
+const { initResources, fetch, closeResources } = require('../utils');
+const { db } = require('../../src/db');
+const { subjectJWT } = require('../../src/utils/jwt');
+const { deployContract } = require('../blockchain-setup');
+const { hash } = require('../../src/utils/encryption');
+const { BadRequest } = require('../../src/utils/errors');
+
+beforeAll(initResources);
+
+afterAll(closeResources);
+
+async function createUser(id) {
+  const token = await subjectJWT.sign({ subjectId: id });
+  await fetch('/api/subject/give-consent', {
+    method: 'POST',
+    body: { personalData: { name: 'test' } },
+    headers: {
+      Authorization: `Bearer ${token}`
+    }
+  });
+  return token;
+}
+
+function fetchWithAuthorization(path, token, overrides) {
```

Comment:
What is the difference between this fetch and the other we were using?

Reply:
It is a wrapper that adds the `Authorization` header. @Jakedavies maybe we should extract it to `utils`, cause it's used all over the place?

Reply:
yeah I think abstracting it makes sense, I wanted to play with it here to see if It saved any time.